File: rust/src/selection_sort.rs
pub fn copy(arr: &[i32]) -> Vec<i32> {
    let mut arr_copy = Vec::from(arr);
    let mut result = Vec::with_capacity(arr.len());

    while arr_copy.len() > 0 {
        let mut min_number = arr_copy[0];
        let mut min_index: usize = 0;

        for (number_index, number) in arr_copy.iter().enumerate() {
            if *number <= min_number {
                min_number = *number;
                min_index = number_index;
            }
        }

        arr_copy.swap_remove(min_index);
        result.push(min_number);
    }

    return result;
}

#[cfg(test)]
mod tests {
    #[test]
    fn selection_sort() {
        let mut arr = vec![24, 1, 525, 27, 75];
        let result = vec![1, 24, 27, 75, 525];
        assert_eq!(crate::selection_sort::copy(&mut arr), result);
    }
}

File: javascript/selectionSort.js
const selectionSort = (arr) => {
  arr = [...arr];
  const result = [];

  while (arr.length > 0) {
    //Find the minimum number
    let minNumber = arr[0];
    let minIndex = 0;
    arr.forEach((number, i) => {
      if (number < minNumber) {
        minNumber = number;
        minIndex = i;
      }
    });

    //Push the minimum value to result array
    result.push(minNumber);

    //Swap the min number to the end of the array and pop it from the array
    arr[minIndex] = arr[arr.length - 1];
    arr[arr.length - 1] = minNumber;
    arr.pop();
  }

  return result;
};

module.exports = selectionSort;
